### DISCUSSION TITLE
CWEB with OpTeX

### DISCUSSION THREAD

**Discussion by omasanori** [olsak/OpTeX#105] [Q&A] [answered]
Hi, I think it would be great if OpTeX can be used with CWEB, though it can't at this moment.

Using the `common.w` example in `cwebman`, `cweave common.w; optex common` generates the following error:
```
This is LuaTeX, Version 1.15.0 (TeX Live 2022)
 restricted system commands enabled.
(./common.tex This is OpTeX (Olsak's Plain TeX), version <1.10 Feb 2023>
(/usr/local/texlive/2022/texmf-dist/tex/plain/cweb/cwebmac.tex
(/usr/local/texlive/2022/texmf-dist/tex/generic/iftex/iftex.sty)
! Undefined control sequence.
l.81 ...PPbox=\hbox{\kern.5pt\raise1pt\hbox{\sevenrm
                                                  +\kern-1pt+}\kern.5pt}
?
```
Some modifications seem (unsurprisingly?) needed to `cwebmac`.

Is there anyone interested in OpTeX + CWEB, or even having tried the modifications?

Update: Well, I realized that we need to modify `cweave` too.

**Comment by olsak** [comment, marked as answer]
The problem is that plain TeX uses ancient 7bits fonts and cwebmac supposes that these fonts are still used in year 2023.
But this is not true when OpTeX is used.

We can do a "conservative patch" where we load these ancient fonts in the same manner as plain.tex does it. Of course, this solution doesn't co-operate with OpTeX's font selection system. Maybe it doesn't matter, maybe a user only want to return to the year 1980.
Then we can do this:

```
\font\sevenrm=cmr7
\font\sevensy=cmsy7
\font\teni=cmmi10
\def\pagebody{\_vbox to\_vsize {\_boxmaxdepth=\_maxdepth \_pagecontents}}% \pagebody in plainTeX

\input cwebmac

Hello

\bye
```
I didn't test any real document. It  is very likely that other problems will appear. We can solve them similarly: to find the control sequence in plain.tex file a define it in the same manner.

If we want more, i.e. use OpTeX features together with cwebmac, then we must to declare more exactly what we want to do and several macros from cwebmac should be rewritten.

**Reply by omasanori** [reply]
Thanks for your detailed explanation!
